lib/performance_platform/gateway/s3_ip_locations.rb

```diff
@@ -0,0 +1,28 @@
+require 'aws-sdk-s3'
+
+class PerformancePlatform::Gateway::S3IpLocations
+  def initialize
+    @s3 = Aws::S3::Client.new(config)
+  end
+
+  def fetch
+    bucket = ENV.fetch('S3_PUBLISHED_LOCATIONS_IPS_BUCKET')
+    key = ENV.fetch('S3_PUBLISHED_LOCATIONS_IPS_OBJECT_KEY')
+
+    resp = s3.get_object(bucket: bucket, key: key)
+    JSON.parse(resp.body.read)
+  end
+
+private
+
+  def config
+    {
+      region: 'eu-west-2',
+      access_key_id: ENV.fetch('ACCESS_KEY_ID'),
```

Comment:
This isn't covered by tests.

I can't see tests covering credentials in the `admin` side either, so are we happy with us delegating responsibility to the gem for handling this all correctly?

Reply:
We don't need Access keys here.
Access policies allow API calls in AWS

Reply:
Huh. If I don't config this, tests blow up - wonder how I write tests that properly reflect what we're doing in AWS.